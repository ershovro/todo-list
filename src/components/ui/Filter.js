import React from 'react';
import PropTypes from 'prop-types';

const FILTER_STATES = ['All', 'Active', 'Completed'];
const resolveClassName = (state, filtered) => (
   `filter__button ${state === filtered ? 'filter__button_selected' : ''}`
);

const Filter = ({filtered = 'All', onFilterButtonClick = f => f}) => (
   <div className="filter">
      Show: { FILTER_STATES.map( (state, i) =>
         <button
            key={i}
            className={ resolveClassName(state, filtered) }
            onClick={ () => onFilterButtonClick( state.toLowerCase() ) }>
            {state}
         </button>
       ) }
   </div>
);

Filter.propTypes = {
   filtered: PropTypes.string,
   onFilterButtonClick: PropTypes.func
};

export default Filter;