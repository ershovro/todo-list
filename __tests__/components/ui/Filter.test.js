import React from 'react';
import Filter from "../../../src/components/ui/Filter";
import enzyme from 'enzyme';
import { JSDOM } from 'jsdom';
import toJSON from 'enzyme-to-json'
import { compose } from 'redux';
import Adapter from 'enzyme-adapter-react-16';


const { window } = new JSDOM(`...`);
const { document } = (new JSDOM(`...`)).window;

enzyme.configure({ adapter: new Adapter() });
global.window = window;
global.document = document;


describe('<Filter /> ui component', () => {

   describe('Rendering ui', () => {

      /*it('Renders Correctly', () => {
         compose(expect, toJSON, enzyme.shallow)(
            <AddTodoForm />
         ).toMatchSnapshot();
      });*/

      it('Renders div with class .filter', () => {
         expect(
            enzyme.mount( <Filter /> ).
            find('div.filter').
               length
         ).toBe(1)
      });

      it('Clicking default does not cause error', () => {
         const wrapper = enzyme.mount(  <Filter /> );

         wrapper.find('.filter__button').simulate('click');
      });

   });

   describe('Filter button click', () => {
      const _click = jest.fn();
      const wrapper = enzyme.mount(  <Filter onFilterButtonClick={_click}/> );

      wrapper.find('div.filter__button').first().simulate('click');

      it('Invoke onSubmit handler', () => {
         expect(_click).toBeCalled();
      });

      it('Invoke onSubmit handler with correct arguments', () => {
         expect(_click).toBeCalledWith('All');
      });

   });
});
