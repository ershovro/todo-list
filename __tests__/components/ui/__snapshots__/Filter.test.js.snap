// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> ui component Rendering ui Renders Correctly 1`] = `
<div
  className="filter"
>
  Show: 
  <button
    className="filter__button filter__button_selected"
    key="0"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="filter__button "
    key="1"
    onClick={[Function]}
  >
    Active
  </button>
  <button
    className="filter__button "
    key="2"
    onClick={[Function]}
  >
    Completed
  </button>
</div>
`;
